### PR TITLE
Hotfix/pointer search paginator

### PR DESCRIPTION
**Pull request by chennan47**
resolve issue: #1695

<b>Purpose</b>
solve the problem that there is no pagination for add link search result
image

<b>Changes</b>
add paginator to add link serach result
image

### CODE REVIEW

File: website/project/views/node.py [hunk 1]
```diff
@@ -1068,7 +1070,11 @@ def search_node(**kwargs):
     auth = kwargs['auth']
     node = Node.load(request.json.get('nodeId'))
     include_public = request.json.get('includePublic')
+    size = float(request.json.get('size', '5').strip())
```
Comment (sloria) [inline website/project/views/node.py]: Shouldn't this always be an `int`?

Comment (sloria) [inline website/project/views/node.py]: Also, if the input is invalid, e.g. the provided `size` is not an int, we should return a proper error response.

File: website/project/views/node.py [hunk 2]
```diff
@@ -1068,7 +1070,11 @@ def search_node(**kwargs):
...
+    page = request.json.get('page', '0')
```
Comment (sloria) [inline website/project/views/node.py]: This should default to `0`, not `'0'`.

File: website/project/views/node.py (hunk 1 above)
Comment (chennan47) [inline website/project/views/node.py]: this is set to float for the later math calculate https://github.com/CenterForOpenScience/osf.io/pull/2075/files#diff-033d4b3df3102b7d1aace074938585e9R1100 
pages = math.ceil(count / size)

File: website/project/views/node.py (hunk 2 above)
Comment (chennan47) [inline website/project/views/node.py]: resolved in  d7e27fe

**Comment by sloria** [comment]
There is a lot of duplicated from `contribAdder.js` here. Can you refactor this to reduce the duplications, perhaps through helper functions and/or subclassing (we now support [ES6 classes](https://babeljs.io/docs/learn-es6/#classes) if module's file extension is `.es6.js`)?

Once that refactor is done, I will do another pass. :point_up:

**Comment by chennan47** [comment]
Using the new es6 to refactor paginator, so now move the base to be develop. The new pull request is https://github.com/CenterForOpenScience/osf.io/pull/2134

**Pull request closed without merging**